sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientTestsLive.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.Testing;
+using NUnit.Framework;
+
+namespace Azure.AI.TextAnalytics.Tests
+{
+    public class TextAnalyticsClientTestsLive : RecordedTestBase
+    {
+        public const string EndpointEnvironmentVariable = "TEXT_ANALYTICS_ENDPOINT";
+        public const string SubscriptionKeyEnvironmentVariable = "TEXT_ANALYTICS_SUBSCRIPTION_KEY";
+
+        public TextAnalyticsClientTestsLive(bool isAsync) : base(isAsync, RecordedTestMode.Live) { }
+
+        public TextAnalyticsClient GetClient()
+        {
+            return InstrumentClient
+                (new TextAnalyticsClient(
+                    new Uri(Recording.GetVariableFromEnvironment(EndpointEnvironmentVariable)),
+                    Recording.GetVariableFromEnvironment(SubscriptionKeyEnvironmentVariable),
+                    Recording.InstrumentClientOptions(new TextAnalyticsClientOptions())));
+        }
+
+        [Test]
+        [Ignore ("_options not set. How to get def language?")]
```

Comment:
@pakrym the recordings use the [protected constructor](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L31) that doesn't set the `TextAnaliticsClientOptions` and in that object TA has 2 default properties (ie [DefaultCountryHit](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientOptions.cs#L40)).

- Is it ok to have those default properties in the `ClientOptions` object?
- if so, is there a way to preserve the `ClientOptions` object when using recording?

Reply:
I'm not sure I understand, `GetClient` uses the public ctor https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L58

Reply:
Looks like it, but when `InstrumentClient` returns, the options object is set to null.
When debugging it, I found that [here ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/TestFramework/ClientTestBase.cs#L77)it ends up calling the [protected ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L31) constructor for TA which doesn't accept ClientOptions.

Reply:
Ah, I see, what you are getting is a "proxy" object that forwards all calls to an inner client. So if you call any virtual method on the proxy it would forward the call to the real client. This means that not having options on the proxy object shouldn't matter because nothing should try to use it.

Are you sure all methods you are calling in test are marked as virtual? What is the exception when you reenable the test?

Reply:
Oh cool! I missed that part of the proxy. Thank you for explaining it :)
Marking the [method ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L1712) as virtual worked